src/librustc_mir/borrow_check/nll/type_check/liveness.rs

```diff
@@ -168,7 +168,10 @@ impl<'gen, 'typeck, 'flow, 'gcx, 'tcx> TypeLivenessGenerator<'gen, 'typeck, 'flo
         );
 
         cx.tcx().for_each_free_region(&value, |live_region| {
-            cx.constraints.liveness_set.push((live_region, location));
+            if let Some(borrowck_context) = cx.borrowck_context {
+                let region_vid = borrowck_context.universal_regions.to_region_vid(live_region);
+                borrowck_context.constraints.liveness_constraints.add_element(region_vid, location);
```

Comment:
Here we can also do 

```rust
if let Some(all_facts) = &mut borrowck_context.all_facts {
    all_facts.region_live_at.push(...);
}
```

and then -- I suspect -- we can remove that big awkward loop above.

Reply:
Fixed this.